nexus/types/src/deployment/blueprint_diff.rs

```diff
@@ -1046,6 +1419,72 @@ impl<'diff> BlueprintDiffDisplay<'diff> {
         ]
     }
 
+    pub fn make_clickhouse_cluster_config_diff_tables(
+        &self,
+    ) -> Option<ClickhouseClusterConfigDiffTables> {
+        match (
+            &self.diff.before_clickhouse_cluster_config,
+            &self.diff.after_clickhouse_cluster_config,
+        ) {
+            // Before collection + after blueprint
+            (
+                DiffBeforeClickhouseClusterConfig::Collection {
+                    latest_keeper_membership: Some(before),
+                    ..
+                },
+                Some(after),
+            ) => Some(ClickhouseClusterConfigDiffTables::diff_collection_and_blueprint(before, after)),
+
+            // Before collection only
+            (
+                DiffBeforeClickhouseClusterConfig::Collection {
+                    latest_keeper_membership: Some(before),
+                    ..
+                },
+                None,
+            ) => Some(ClickhouseClusterConfigDiffTables::removed_from_collection(before)),
```

Comment:
W.r.t. my question earlier about the unwrap; I think the situation is:

* The only method that doesn't fill in `ClickhouseClusterConfigDiffTables::servers` is `removed_from_collection()`
* `make_clickhouse_cluster_config_diff_tables()` can therefore also return a `servers` value of `None`

I guess we could make a second diff table type where `servers` is not optional, return it from all the methods except this one and `removed_from_collection()`, and add a `From` impl to convert it into `ClickhouseClusterConfigDiffTables`? Doesn't seem terrible, but does seem like a fair bit of busywork to avoid a single `.unwrap()`. I'd probably do it, but if you'd rather not that's fine with me.

Reply:
Thanks for the suggestion @jgallagher. I was unhappy with this unwrap as well, especially since the SAFETY message would become invalid if someone changed a method far away from that message.

Fix made in [81e85f5](https://github.com/oxidecomputer/omicron/pull/6968/commits/81e85f58150c5392dd379240caa01590ea271799)